build-tools/create-pkg/scripts/preinstall

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+INSTALLATION_ROOT=/Library/Frameworks/Xamarin.Android.framework/Versions
+PREVIOUS_VERSION=$(readlink "$INSTALLATION_ROOT/Current")
+PREVIOUS_VERSION_PATH="$INSTALLATION_ROOT/$PREVIOUS_VERSION"
```

Comment:
One possible problem here is that it assumes that `Current` is a relative symlink into the same directory.  It's possible (rare! but possible) that `Current` could instead be a symlink into e.g. a xamarin-android or monodroid build tree, which is an idea we occasionally throw around for "how do I get Visual Studio for Mac to use my local build tree for testing?".

If this is the case, we don't want to nuke that. :-)

I *think* that this might work:

```bash
if echo "$PREVIOUS_VERSION" | grep '/\|\.\.' >/dev/null 2>& ; then
	# Appears to be a relative or absolute path; ignore!
	exit 0;
fi
```

Reply:
I think we're ok with this as is. In the case mentioned above we'd just end up constructing a bogus path (e.g. `/Library/Frameworks/Xamarin.Android.framework/Versions//Users/user/source/xamarin-android/bin/Debug`). The subsequent directory test would fail and no delete would occur.